src/components/tooltip/tooltip.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FC, HTMLAttributes, useState, ReactNode } from 'react'
+import { ElToolTipChild, ElToolTipContainer, ElToolTipLabel } from './styles'
+import { useId } from '#src/storybook/random-id'
+
+export interface ToolTipProps extends HTMLAttributes<HTMLDivElement> {
+  children: ReactNode
+  label?: string
+  tip: string
+  maxWidth?: string
+  position?:
+    | 'top'
+    | 'bottom'
+    | 'right'
+    | 'left'
+    | 'topStart'
+    | 'topEnd'
+    | 'bottomStart'
+    | 'bottomEnd'
+    | 'rightStart'
+    | 'rightEnd'
+    | 'leftStart'
+    | 'leftEnd'
+}
+
+export interface ToolTipChildProps extends HTMLAttributes<HTMLDivElement> {
+  position: ToolTipProps['position']
+  maxWidth?: string
+}
+
+export const ToolTip: FC<ToolTipProps> = ({ children, label, tip, maxWidth = '400px', position = 'top' }) => {
+  const [visible, setVisible] = useState(false)
+  const tooltipId = useId()
+
+  return (
+    <ElToolTipContainer
+      onMouseEnter={() => setVisible(true)}
+      onMouseLeave={() => setVisible(false)}
+      onFocus={() => setVisible(true)}
+      onBlur={() => setVisible(false)}
+      aria-describedby={tooltipId}
+    >
+      {children}
+      {visible && (
```

Comment:
**issue:** We should really be animating the tooltip in and out, so I think the visibility is best handled in CSS via a `data-visible` attribute.

In future, when we upgrade to React 19 (hopefully we can do this soon), we'll get access to the new [Popover API](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API), which will allow us to push the visibility into the DOM (e.g. https://codesandbox.io/p/sandbox/jp8njk)

Reply:
@kurtdoherty The rationale behind this approach is to address scenarios where the table contains many items (e.g., 100 rows), each requiring a tooltip. Rendering all tooltip data directly in the DOM upfront avoids the need to dynamically generate content on hover dynamically, ensuring better performance and reducing potential rendering delays.
I am happy to discuss this and get it updated to be handled by CSS.

Reply:
@kurtdoherty The rationale behind this approach is to address scenarios where the table contains a large number of items (e.g., 100 rows), each requiring a tooltip. Using conditional rendering ensures better performance and saves memory as we wont' load unnecessary tooltip.

I am happy to discuss and get it updated to be handled by CSS

Reply:
> Using conditional rendering ensures better performance and saves memory as we wont' load unnecessary tooltip.

Have you profiled this? I only ask because I wouldn't have expected there to be much difference at the moment 🤔 If the tooltip was measuring its trigger's bounding box or something like that, then I could see there being a noticeable performance impact, but it seems like a pretty cheap element for browsers to render 🤷‍♂️ 

If performance does become an issue in future, there's CSS properties (e.g. [content-visibility](https://developer.mozilla.org/en-US/docs/Web/CSS/content-visibility)) that could potentially help us either way.

I don't mind having this go through as is. It may actually be possible we can just as easily animate the entry/exit of these DOM elements via [@starting-style](https://developer.mozilla.org/en-US/docs/Web/CSS/@starting-style) regardless of whether we're handling visibility in React component state or CSS (I'd forgotten about this new @-rule, though it's not supported by Firefox, so maybe animation needs to wait either way 🤷)